component/TodoApp.js

```diff
@@ -0,0 +1,68 @@
+import { TodoList } from './TodoList.js';
+import { TodoInput } from './TodoInput.js';
+import { TodoTotalCount } from './TodoTotalCount.js'
+
+export function TodoApp($div) {
+    const $ul = $div.querySelector('#todo-list')
+
+    this.todoItems = [];
+    this.filter = '전체보기';
+
+    this.todoInput = new TodoInput(this);
+    this.todoList = new TodoList($ul, this);
+    this.todoTotalCount = new TodoTotalCount( $div, this );
+
+    this.saveItem = (item) => {
+      this.todoItems.push(item)
+      this.filterTodo(this.filter);
+    }
+
+    this.complete = (todoItem) => {
+      this.todoItems.filter(item => item.todoItem === todoItem)
+                    .forEach(item => item.completed = !item.completed)
```

Comment:
체이닝 연산자를 연속으로 사용하기 보다는 적절한 메서드를 활용해 작성하시면 더 깔끔하고 이해하기 좋은 코드가 될거 같습니다!
```suggestion
      const completeTodos = this.todoItems.filter(item => item.todoItem === todoItem)
      completeTodos.map(item => item.completed = !item.completed)
```

Reply:
좋은 제안 감사합니다. ! 한가지 궁금한 사항이 있어 문의드려요.
자바언어에서는 이와 비슷한 api를 사용할 때 항상 체이닝방식으로 구현을 했었는데요.
```suggestion
      this.todoItems.filter(item => item.todoItem === todoItem)
                    .forEach(item => item.completed = !item.completed)
     this.todoItems.filter(item => item.todoItem === todoItem)
                    .map(item => item.completed = !item.completed)
```
혹시 위와 같은 체이닝 방식은 자바스크립트 내에서는 지양하는지 알 수 있을까요 ?